src/MagnumPlugins/BasisImporter/BasisImporter.h

```diff
@@ -0,0 +1,160 @@
+#ifndef MAGNUM_BASISIMPORTER_BUILD_STATIC
+#ifdef BasisImporter_EXPORTS
+    #define MAGNUM_BASISIMPORTER_EXPORT CORRADE_VISIBILITY_EXPORT
+#else
+    #define MAGNUM_BASISIMPORTER_EXPORT CORRADE_VISIBILITY_IMPORT
+#endif
+#else
+#define MAGNUM_BASISIMPORTER_EXPORT CORRADE_VISIBILITY_STATIC
+#endif
+#define MAGNUM_BASISIMPORTER_LOCAL CORRADE_VISIBILITY_LOCAL
+#else
+#define MAGNUM_BASISIMPORTER_EXPORT
+#define MAGNUM_BASISIMPORTER_LOCAL
+#endif
+
+namespace Magnum { namespace Trade {
+
+/**
+@brief Type to transcode to
+*/
+enum class BasisTranscodingType : Int {
+    Etc1 = 0, /**< ETC1. Loaded as ETC2, prefer ETC2 for better quality. */
+    Etc2 = 1, /**< ETC2 */
+    Bc1 = 2, /**< BC1 */
+    Bc3 = 3, /**< BC3 */
+    Bc4 = 4, /**< BC4 */
+    Bc5 = 5, /**< BC5 */
+    Bc7M6OpaqueOnly = 6, /**< BC7 M4 opaque only */
+    Pvrtc1_4OpaqueOnly = 7 /**< PVRTC1 4 opaque only */
+};
+
+/**
+@brief BASIS importer plugin
+
+@m_keywords{BasisImporterEct1 BasisImporterEtc2 BasisImporterBc1 BasisImporterBc3}
+@m_keywords{BasisImporterBc4 BasisImporterBc5 BasisImporterBc7M6OpaqueOnly}
+@m_keywords{BasisImporterPvrtc1_4OpaqueOnly}
+
+Supports Basis Universal (`*.basis`) compressed images.
+
+This plugin depends on the @ref Trade and [basis_universal](https://github.com/binomialLLC/basis_universal)
+libraries and is built if `WITH_BASISIMPORTER` is enabled when building Magnum
+Plugins. To use as a dynamic plugin, you need to load the @cpp "BasisImporter" @ce
+plugin from `MAGNUM_PLUGINS_IMPORTER_DIR`. To use as a static plugin or as a
+dependency of another plugin with CMake, you need to request the `BasisImporter`
+component of the `MagnumPlugins` package and link to the `MagnumPlugins::BasisImporter`
+target. See @ref building-plugins, @ref plugins for more information.
+
+This plugin provides `BasisImporterEct1`, `BasisImporterEtc1`, `BasisImporterEtc2`,
+`BasisImporterBc1`, `BasisImporterBc3`, `BasisImporterBc4`, `BasisImporterBc5`,
+`BasisImporterBc7M6OpaqueOnly`, `BasisImporterPvrtc1_4OpaqueOnly`.
+
+@section Trade-BasisImporterImporter-configuration Plugin-specific configuration
+
+Basis allows configuration of the format of loaded compressed data.
+The full form of the configuration is shown below:
+
+@snippet MagnumPlugins/BasisImporter/BasisImporter.conf configuration
+
+@subsection Trade-BasisImporterImporter-target-format Target format
+
+Basis is a compressed format that is *transcoded* into a compressed GPU format. With
+BasisImporter, this format can be chosen in different ways:
+
+@snippet MagnumPluginsBasisImporter.cpp basis-target-format-suffix
+
+The list of valid suffixes is equivalent to enum value names in @ref BasisTranscodingType.
+
+If you want to be able to change the target format dynamically, you may want to
+set the "format" configuration of the plugin:
+
+@snippet MagnumPluginsBasisImporter.cpp basis-target-format-config
+
+If you link to the plugin you may also use @ref BasisImporter::setTargetFormat().
+
+*/
```

Comment:
The documentation could be improved:

- Explicitly saying what this is for -- parsing a basis file and giving one of the GPU-specific *compressed* formats on the output. And because there isn't any "one size fits all", one first *has to* explicitly check for support first, and only then then choose a particular format to transcode.
- Referencing (`@ref`) the concrete `CompressedPixelFormat` values for each `BasisTranscodingType`, because those then link to concrete values for GL and Vulkan together with extension information. Without that, the users would have to guess what format will the result be in and what extensions to check for.
- Mentioning the license info like all other plugins have, to be listed at https://doc.magnum.graphics/magnum/credits-third-party.html

Reply:
I am unsure whether I fulfilled point 1 sufficiently, but the improvement on `BasisTranscodingType` is super valuable! And point 3 is done also ✔️

Reply:
Just saw that, wonderful. And it something turns out to be missing, we can always improve later :)